Sources/eppo/Poller.swift

```diff
@@ -0,0 +1,177 @@
+    }
+    
+    func schedule(deadline: TimeInterval, callback: @escaping () async -> Void) {
+        executeCount += 1
+        
+        if isRunning {
+            Task {
+                // Convert deadline to nanoseconds and wait
+                let delayNanos = UInt64(deadline * 1_000_000_000) // Convert seconds to nanoseconds
+                try? await Task.sleep(nanoseconds: delayNanos)
+                
+                if self.isRunning {
+                    await callback()
+                }
+            }
+        }
+    }
+    
+    func cancel() {
+        isRunning = false
+    }
+}
+
+public class Poller {
+    private let intervalMs: Int
+    private let jitterMs: Int
+    private let callback: () async throws -> Void
+    
+    private var stopped = true
+    private var failedAttempts = 0
+    private var nextPollMs: Int
+    private var logger: PollerLogger
+    private let timer: TimerType
+    
+    public init(
+        intervalMs: Int,
+        jitterMs: Int,
+        callback: @escaping () async throws -> Void
+    ) {
+        self.intervalMs = intervalMs
+        self.jitterMs = jitterMs
+        self.callback = callback
+        self.nextPollMs = intervalMs
+        self.logger = PollerLogger()
+        self.timer = RealTimer()
+    }
+    
+    // Internal initializer for testing
+    init(
+        intervalMs: Int,
+        jitterMs: Int,
+        callback: @escaping () async throws -> Void,
+        logger: PollerLogger,
+        timer: TimerType
+    ) {
+        self.intervalMs = intervalMs
+        self.jitterMs = jitterMs
+        self.callback = callback
+        self.nextPollMs = intervalMs
+        self.logger = logger
+        self.timer = timer
+    }
+    
+    public func start() async throws {
+        stopped = false
+        
+        do {
+            try await callback()
+            logger.info("Eppo SDK successfully requested initial configuration")
+        } catch {
+            // Still schedule the first poll even if initial call fails
+        }
+        
+        schedulePoll()
+    }
+    
+    public func stop() {
+        if !stopped {
+            stopped = true
+            timer.cancel()
+            logger.info("Eppo SDK polling stopped")
+        }
+    }
+    
+    private func schedulePoll() {
+        let nextInterval = nextPollMs + randomJitterMs()
+        timer.schedule(deadline: TimeInterval(nextInterval) / 1000.0) { [self] in
+            Task {
+                await poll()
+            }
+        }
+    }
+    
+    private func poll() async {
+        if stopped {
+            return
+        }
+        
+        do {
+            try await callback()
+            failedAttempts = 0
+            nextPollMs = intervalMs
+        } catch {
+            failedAttempts += 1
+            if failedAttempts < PollerConstants.DEFAULT_MAX_POLL_RETRIES {
+                let failureWaitMultiplier = pow(2.0, Double(failedAttempts))
```

Comment:
Exponential! https://github.com/Eppo-exp/js-sdk-common/blob/fb7c554f11f16cbba08f752c06cb9bc46d82f142/src/poller.ts#L124

Reply:
Exponential backoff is life